### PR TITLE
 Revert OpenBSD `buildtype` back to `debugoptimized`

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

After #1910 we might as well see whether the OpenBSD UAFs are still there on an optimized build.

**Test plan**

The OpenBSD build is green, and remains green.

**Closing issues**

...

### CODE REVIEW

File: .builds/openbsd.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ tasks:
         # https://todo.sr.ht/~sircmpwn/builds.sr.ht/274
         ln -s ${HOME}/rizin-testbins test/bins
         # Running the unit tests
-        MALLOC_OPTIONS=S ninja -C build test
+        ninja -C build test
```
Comment (thestr4ng3r) [inline .builds/openbsd.yml]: Better not disable the `MALLOC_OPTIONS` imo. They are nice for detecting bugs that are invisible on other platforms.

**Review by thestr4ng3r — CHANGES_REQUESTED** [review]

File: .builds/openbsd.yml (hunk 1 above)
Comment (kazarmy) [reply .builds/openbsd.yml]: Ok ddb7d56fe1fc1dd2e8ef70085945858dbd51e18d but there won't be any speedup

**Review by thestr4ng3r — APPROVED** [review]

**Comment by wargio** [comment]
can we have a check like `if [ $ASAN == 1]; then MALLOC_OPTIONS=S ninja -C build test ; else ninja -C build test ; fi` ?

**Comment by kazarmy** [comment]
OpenBSD doesn't support ASan afaik

**Comment by kazarmy** [comment]
OpenBSD doesn't support ASan afaik, so it has only 1 CI build

**Comment by kazarmy** [comment]
Sorry what I meant to say is that OpenBSD isn't a major OS so there's only 1 CI build

**Pull request merged**